packages/samples/headless-react/src/components/context/context.ts

```diff
@@ -1,8 +1,11 @@
-import {buildContext} from '@coveo/headless';
-import {engine} from '../../engine';
+import {buildContext, HeadlessEngine, searchAppReducers} from '@coveo/headless';
 
-export function setContext(ageGroup: string, interests: string[]) {
-  buildContext(engine).set({
+export function setContext(
+  engine: HeadlessEngine<typeof searchAppReducers>,
```

Comment:
Since we can no longer read the engine statically, it must be passed here as a parameter.

Additionally, since this isn't a functional component, the `useContext` hook cannot be used as an alternative.